terraform/modules/gro_ingestion_service/lambdas/src/functions/publish-event.function.ts

```diff
@@ -0,0 +1,105 @@
+import { DynamoDBStreamEvent, Handler } from "aws-lambda"
+import { mapToEventRecord } from "../models/EventRecord"
+import { request, RequestOptions } from "https"
+import { PublishEvent } from "../models/PublishEvent"
+import { AttributeValue } from "aws-lambda/trigger/dynamodb-stream"
+
+const apiUrl = process.env.API_URL
+const authUrl = process.env.AUTH_URL
+const clientId = process.env.CLIENT_ID
+const clientSecret = process.env.CLIENT_SECRET
+
+const makeRequest = async (options: RequestOptions, requestData: string): Promise<object> => new Promise((resolve, reject) => {
```

Comment:
I'd be interested to have _some_ form of testing around these, as a bit of a framework for how we test more complicated things